opensoundscape/datasets.py

```diff
@@ -178,138 +179,174 @@ def collate_fn(*batch):
-
-        image = Image.fromarray(spectrogram.spectrogram.astype(np.uint8),mode='L')
-        
-        #add a blended/overlayed image from another class directly on top
+            audio, audio_length = self.random_audio_trim(audio, audio_length)
+        image = self.image_from_audio(audio)
+
+        # add a blended/overlayed image from another class directly on top
         if self.overlay_prob > np.random.uniform():
-            
-            # select a random training file from a different class and create spectrogram
-            this_class = row[self.label_column]
-            other_classes_df = self.df[self.df[self.label_column]!=this_class]
-            file_path = np.random.choice(other_classes_df[self.audio_column].values)
-            overlay_audio = Audio.from_file(file_path)
-            overlay_spectrogram = Spectrogram.from_audio(overlay_audio)
-            overlay_spectrogram = overlay_spectrogram.linear_scale(feature_range=(0, 255))
-            
-            # trim to same length as main clip
-            overlay_audio_length = len(overlay_audio.samples)/overlay_audio.sample_rate
-            if overlay_audio_length < audio_length:
-                raise ValueError(f'the length of the overlay file ({overlay_audio_length} sec) was less than the length of the file {file_path} ({audio_length} sec)')
-            elif overlay_audio_length > audio_length:
-                extra_time = audio_length - overlay_audio_length
-                start_time = np.random.uniform()*extra_time
-                overlay_spectrogram = overlay_spectrogram.trim(start_time,start_time+audio_length)
-                
-            # create an image and add blur
-            overlay_image = Image.fromarray(overlay_spectrogram.spectrogram.astype(np.uint8),mode='L')
-            blur_r = np.random.randint(0, 8) / 10
-            overlay_image = overlay_image.filter(ImageFilter.GaussianBlur(radius=blur_r))
-
-            # use a weighted sum to overlay (blend) the images
-            overlay_weight = np.random.randint(2, 5) / 10 #<0.5 means more emphasis on original image
-            image = Image.blend(image,overlay_image,overlay_weight)
-
-        image = image.convert("RGB")
-
-                
+            image = self.overlay_random_image(
+                original_image=image, original_length=audio_length
+            )
+
         if self.debug:
-            image.save(f"{self.debug}/{audio_p.stem}.png")
+            image.save(f"{self.debug}/{audio_path.stem}_{time()}.png")
 
+        # apply desired random transformations to image and convert to tensor
         X = self.transform(image)
 
-        if self.spec_augment:
-            X = X[0].unsqueeze(0)
-
+        # Return data : label pairs (training/validation)
         if self.label_column:
-            labels = np.array([row[self.label_column]])
+            labels = np.array([self.row[self.label_column]])
```

Comment:
Why save this in the Dataset?

Reply:
Fixed by https://github.com/kitzeslab/opensoundscape/pull/179/commits/bbfccac7e1e6eb9f73583f861c0720a2c1f45cc7